showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITApiVersionHeaders.java

```diff
@@ -0,0 +1,311 @@
+  private GrpcCapturingClientInterceptor grpcComplianceInterceptor;
+  private EchoClient grpcClient;
+  private EchoClient httpJsonClient;
+  private ComplianceClient grpcComplianceClient;
+  private ComplianceClient httpJsonComplianceClient;
+
+  @Before
+  public void createClients() throws Exception {
+    // Create gRPC Interceptor and Client
+    grpcInterceptor = new GrpcCapturingClientInterceptor();
+    grpcClient = TestClientInitializer.createGrpcEchoClient(ImmutableList.of(grpcInterceptor));
+
+    // Create HttpJson Interceptor and Client
+    httpJsonInterceptor = new HttpJsonCapturingClientInterceptor();
+    httpJsonClient =
+        TestClientInitializer.createHttpJsonEchoClient(ImmutableList.of(httpJsonInterceptor));
+
+    // Create gRPC ComplianceClient and Interceptor
+    // Creating a compliance client to test case where api version is not set
+    grpcComplianceInterceptor = new GrpcCapturingClientInterceptor();
+    grpcComplianceClient =
+        TestClientInitializer.createGrpcComplianceClient(
+            ImmutableList.of(grpcComplianceInterceptor));
+
+    // Create HttpJson ComplianceClient and Interceptor
+    httpJsonComplianceInterceptor = new HttpJsonCapturingClientInterceptor();
+    httpJsonComplianceClient =
+        TestClientInitializer.createHttpJsonComplianceClient(
+            ImmutableList.of(httpJsonComplianceInterceptor));
+  }
+
+  @After
+  public void destroyClient() {
+    grpcClient.close();
+    httpJsonClient.close();
+    grpcComplianceClient.close();
+    httpJsonComplianceClient.close();
+  }
+
+  @Test
+  public void testGrpc_matchesApiVersion() {
+    grpcClient.echo(EchoRequest.newBuilder().build());
+    String headerValue = grpcInterceptor.metadata.get(API_VERSION_HEADER_KEY);
+    assertThat(headerValue).isEqualTo(EXPECTED_ECHO_API_VERSION);
+  }
+
+  @Test
+  public void testHttpJson_matchesHeaderName() {
+    httpJsonClient.echo(EchoRequest.newBuilder().build());
+    ArrayList headerValues =
+        (ArrayList) httpJsonInterceptor.metadata.getHeaders().get(HTTP_RESPONSE_HEADER_STRING);
+    String headerValue = (String) headerValues.get(0);
+    assertThat(headerValue).isEqualTo(EXPECTED_ECHO_API_VERSION);
+  }
+
+  @Test
+  public void testGrpc_noApiVersion() {
+    RepeatRequest request =
+        RepeatRequest.newBuilder().setInfo(ComplianceData.newBuilder().setFString("test")).build();
+    grpcComplianceClient.repeatDataSimplePath(request);
+    assertThat(API_VERSION_HEADER_KEY).isNotIn(grpcComplianceInterceptor.metadata.keys());
+  }
+
+  @Test
+  public void testHttpJson_noApiVersion() {
+    RepeatRequest request =
+        RepeatRequest.newBuilder().setInfo(ComplianceData.newBuilder().setFString("test")).build();
+    httpJsonComplianceClient.repeatDataSimplePath(request);
+    assertThat(API_VERSION_HEADER_KEY)
+        .isNotIn(httpJsonComplianceInterceptor.metadata.getHeaders().keySet());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
```

Comment:
Can you use `assertThrows()` instead of `@Test(expected = ...)`? Something like: https://github.com/googleapis/sdk-platform-java/blob/3611e8f8982070f6b1aa14b6272677608aa5f6fb/showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITUnaryCallable.java#L70-L71